src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,29 +1,38 @@
 package mmosii.bookstore.service.impl;
 
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mmosii.bookstore.dto.BookDto;
+import mmosii.bookstore.dto.CreateBookRequestDto;
+import mmosii.bookstore.exception.EntityNotFoundException;
+import mmosii.bookstore.mapper.BookMapper;
 import mmosii.bookstore.model.Book;
 import mmosii.bookstore.repository.BookRepository;
 import mmosii.bookstore.service.BookService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.fromDto(requestDto)));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
-
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(()
+                -> new EntityNotFoundException("Can't find book by Id " + id));
+        return bookMapper.toDto(book);
```

Comment:
you can do this via single Optional methods chain

don't forget that Optional has a `map` method :)